azext_edge/edge/providers/support/clusterconfig/billing.py

```diff
@@ -0,0 +1,126 @@
+# coding=utf-8
+# ----------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License file in the project root for license information.
+# ----------------------------------------------------------------------------------------------
+
+from functools import partial
+from typing import Iterable
+
+from knack.log import get_logger
+
+from ...edge_api import CLUSTER_CONFIG_API_V1, EdgeResourceApi
+from ..base import (
+    DAY_IN_SECONDS,
+    assemble_crd_work,
+    process_cron_jobs,
+    process_deployments,
+    process_jobs,
+    process_replicasets,
+    process_services,
+    process_v1_pods,
+)
+
+logger = get_logger(__name__)
+
+AIO_BILLING_USAGE_NAME_LABEL = "app.kubernetes.io/name in (microsoft-iotoperations)"
+ARC_BILLING_EXTENSION_COMP_LABEL = "app.kubernetes.io/component in (billing-operator)"
+BILLING_RESOURCE_KIND = "billing"
+
+
+def fetch_pods(
+    since_seconds: int = DAY_IN_SECONDS,
+):
+    # capture billing pods for aio usage
+    billing_pods = process_v1_pods(
+        resource_api=CLUSTER_CONFIG_API_V1,
+        label_selector=AIO_BILLING_USAGE_NAME_LABEL,
+        since_seconds=since_seconds,
+        sub_group=BILLING_RESOURCE_KIND,
+    )
+
+    # capture billing pods for arc extension
+    billing_pods.extend(
+        process_v1_pods(
+            resource_api=CLUSTER_CONFIG_API_V1,
+            label_selector=ARC_BILLING_EXTENSION_COMP_LABEL,
+            since_seconds=since_seconds,
+            sub_group=BILLING_RESOURCE_KIND,
+        )
+    )
+
+    return billing_pods
+
+
+def fetch_jobs():
+    processed = process_jobs(
+        resource_api=CLUSTER_CONFIG_API_V1,
+        label_selector=AIO_BILLING_USAGE_NAME_LABEL,
+        sub_group=BILLING_RESOURCE_KIND,
+    )
+
+    return processed
+
+
+def fetch_cron_jobs():
+    processed = process_cron_jobs(
+        resource_api=CLUSTER_CONFIG_API_V1,
+        label_selector=AIO_BILLING_USAGE_NAME_LABEL,
+        sub_group=BILLING_RESOURCE_KIND,
+    )
+
+    return processed
+
+
+def fetch_deployments():
+    processed = process_deployments(
+        resource_api=CLUSTER_CONFIG_API_V1,
+        label_selector=ARC_BILLING_EXTENSION_COMP_LABEL,
+        sub_group=BILLING_RESOURCE_KIND,
+    )
+
+    return processed
+
+
+def fetch_replicasets():
+    processed = []
+    processed.extend(process_replicasets(
```

Comment:
Curious what the extend is for when there is no other data in processed [ ]. We can return function output directly, or assign to processed and return processed (such as in `fetch_deployments` higher up).